src/DateInput/index.js

```diff
@@ -89,6 +90,33 @@ class DateInput extends React.Component {
     this.handlePresetChange = this.handlePresetChange.bind(this)
   }
 
+  componentDidUpdate (prevProps) {
+    const {
+      dates: {
+        end: receivedEnd,
+        start: receivedStart,
+      },
+    } = this.props
+
+    if (equals(prevProps, this.props) || !receivedStart || !receivedEnd) return
+
+    const {
+      dates: {
+        end: currentEnd,
+        start: currentStart,
+      },
+    } = this.state
+
+    const isSameStart = receivedStart.isSame(currentStart, 'day')
```

Comment:
I tried to modify the code a bit to remove the first if in line 101, my suggestion is to verify the conditions in a single if in line 113. What do you think?
```suggestion
    const isSameStart = receivedStart && receivedStart.isSame(currentStart, 'day')
    const isSameEnd = receivedEnd && receivedEnd.isSame(currentEnd, 'day')

    if (!equals(prevProps, this.props) && (!isSameStart || !isSameEnd)) {
      this.setState({ // eslint-disable-line react/no-did-update-set-state
        dates: momentToText(this.props.dates),
      })
    }
```

Reply:
Nice! It is much cleaner this way